Content.Client/Lathe/UI/LatheMenu.xaml.cs

```diff
@@ -61,30 +67,36 @@ public LatheMenu(LatheBoundUserInterface owner)
         }
     }
 
-    public void PopulateMaterials(EntityUid lathe)
+    public void PopulateMaterials(EntityUid latheUid)
     {
-        if (!_entityManager.TryGetComponent<MaterialStorageComponent>(lathe, out var materials))
+        if (!_entityManager.TryGetComponent<MaterialStorageComponent>(latheUid, out var materials))
+            return;
+        if (!_entityManager.TryGetComponent<LatheComponent>(latheUid, out var lathe))
             return;
 
-        Materials.Clear();
+        Materials.DisposeAllChildren();
 
         foreach (var (id, amount) in materials.Storage)
         {
             if (!_prototypeManager.TryIndex(id, out MaterialPrototype? material))
                 continue;
             var name = Loc.GetString(material.Name);
-            var mat = Loc.GetString("lathe-menu-material-display",
-                ("material", name), ("amount", amount));
-            Materials.AddItem(mat, _spriteSystem.Frame0(material.Icon), false);
-        }
 
-        if (Materials.Count == 0)
-        {
-            var noMaterialsMsg = Loc.GetString("lathe-menu-no-materials-message");
-            Materials.AddItem(noMaterialsMsg, null, false);
```

Comment:
now the message won't display if there are no materials

Reply:
@AJCM-git Why do we need this message? Isn't it obvious that if it has 0, you're clicking eject and nothing will be ejected?

Reply:
It's important player feedback, having an empty UI block does not communicate the purpose of that part of the UI as well as a "No Materials" message